src/mlpack/bindings/go/mlpack/arma_util.go

```diff
@@ -0,0 +1,396 @@
+  e, err := m.Dims()
+  if err != 1 {
+  panic("Must have a single column")
+  }
+
+  blas64General := m.RawMatrix()
+  data := blas64General.Data
+
+  // Pass pointer of the underlying matrix to Mlpack.
+  ptr := unsafe.Pointer(&data[0])
+  C.mlpackToArmaUrow(C.CString(identifier), (*C.double)(ptr), C.int(e))
+}
+
+// GonumToArma passes a gonum matrix to C by using it's gonums underlying blas64.
+func gonumToArmaCol(identifier string, m *mat.Dense) {
+  // Get matrix dimension, underlying blas64General matrix, and data.
+  err, e := m.Dims()
+  if err != 1 {
+  panic("Must have a single row")
+  }
+
+  blas64General := m.RawMatrix()
+  data := blas64General.Data
+
+  // Pass pointer of the underlying matrix to Mlpack.
+  ptr := unsafe.Pointer(&data[0])
+  C.mlpackToArmaCol(C.CString(identifier), (*C.double)(ptr), C.int(e))
+}
+
+// GonumToArma passes a gonum matrix to C by using it's gonums underlying blas64.
+func gonumToArmaUcol(identifier string, m *mat.Dense) {
+  // Get matrix dimension, underlying blas64General matrix, and data.
+  err, e := m.Dims()
+  if err != 1 {
+  panic("Must have a single row")
+  }
+
+  blas64General := m.RawMatrix()
+  data := blas64General.Data
+
+  // Pass pointer of the underlying matrix to Mlpack.
+  ptr := unsafe.Pointer(&data[0])
+  C.mlpackToArmaUcol(C.CString(identifier), (*C.double)(ptr), C.int(e))
+}
+
+// GonumToArmaMatWithInfo passes a gonum matrix with info to C by 
+// using it's gonums underlying blas64.
+func gonumToArmaMatWithInfo(identifier string, m *DataWithInfo) {
+  // Get matrix dimension, underlying blas64General matrix, and data.
+  r, c := m.Data.Dims()
+  blas64General := m.Data.RawMatrix()
+  DataAndInfo := blas64General.Data
+  boolarray := m.Cat
+  // Pass pointer of the underlying matrix to Mlpack.
+  boolptr := unsafe.Pointer(&boolarray[0])
+  matptr := unsafe.Pointer(&DataAndInfo[0])
+  C.mlpackToArmaMatWithInfo(C.CString(identifier), (*C.bool)(boolptr),
+      (*C.double)(matptr), C.size_t(c), C.size_t(r))
+}
+
+// ArmaToGonum returns a gonum matrix based on the memory pointer
+// of an armadillo matrix.
+func (m *mlpackArma) armaToGonumMat(identifier string) *mat.Dense {
+  // Armadillo row and col
+  c := int(C.mlpackNumRowMat(C.CString(identifier)))
+  r := int(C.mlpackNumColMat(C.CString(identifier)))
+  e := int(C.mlpackNumElemMat(C.CString(identifier)))
+
+  // Allocate Go memory pointer to the armadillo matrix.
+  m.allocArmaPtrMat(identifier)
+  runtime.GC()
+  time.Sleep(time.Second)
+
+  // Convert pointer to slice of data, to then pass it to a gonum matrix.
+  array := (*[1<<30 - 1]float64)(m.mem)
+  if array != nil {
+    data := array[:e]
+
+    // Initialize result matrix.
+    output := mat.NewDense(r, c, data)
+
+    // Return gonum vector.
+    return output
+  }
+  return nil
+}
+
+// ArmaToGonum returns a gonum matrix based on the memory pointer
+// of an armadillo matrix.
+func (m *mlpackArma) armaToGonumArray(identifier string) (int, int, []float64) {
```

Comment:
This is a general style comment, but I think maybe we should try to match the mlpack style as closely as we can.  So, e.g.,

```
func (m *mlpackArma) armaToGonumArray(identifier string) (int, int, []float64)
{
```

Reply:
Sorry This will not work. Go doesn't allow this style.

Reply:
Huh.  I hadn't known.  https://stackoverflow.com/questions/17153838/why-does-golang-enforce-curly-bracket-to-not-be-on-the-next-line
So I guess we are stuck with this style, but that's okay. :)